com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDGpuLightsBuilder.Jobs.cs

```diff
@@ -0,0 +1,910 @@
+            [ReadOnly]
+            public NativeArray<HDProcessedVisibleLight> processedEntities;
+            [ReadOnly]
+            public NativeArray<VisibleLight> visibleLights;
+            [ReadOnly]
+            public NativeArray<LightBakingOutput> visibleLightBakingOutput;
+            [ReadOnly]
+            public NativeArray<LightShadowCasterMode> visibleLightShadowCasterMode;
+            #endregion
+
+            #region output processed lights
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<LightData> lights;
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<DirectionalLightData> directionalLights;
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<LightsPerView> lightsPerView;
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<SFiniteLightBound> lightBounds;
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<LightVolumeData> lightVolumes;
+            [WriteOnly]
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<int> gpuLightCounters;
+            #endregion
+
+            private ref HDLightRenderData GetLightData(int dataIndex)
+            {
+#if DEBUG
+                if (dataIndex < 0 || dataIndex >= totalLightCounts)
+                    throw new Exception("Trying to access a light from the DB out of bounds. The index requested is: " + dataIndex + " and the length is " + totalLightCounts);
+#endif
+                unsafe
+                {
+                    HDLightRenderData* data = (HDLightRenderData*)lightRenderDataArray.GetUnsafePtr<HDLightRenderData>() + dataIndex;
+                    return ref UnsafeUtility.AsRef<HDLightRenderData>(data);
+                }
+            }
+
+            private static uint GetLightLayer(bool lightLayersEnabled, in HDLightRenderData lightRenderData)
+            {
+                int lightLayerMaskValue = (int)lightRenderData.lightLayer;
+                uint lightLayerValue = lightLayerMaskValue < 0 ? (uint)LightLayerEnum.Everything : (uint)lightLayerMaskValue;
+                return lightLayersEnabled ? lightLayerValue : uint.MaxValue;
+            }
+
+            private static Vector3 GetLightColor(in VisibleLight light) => new Vector3(light.finalColor.r, light.finalColor.g, light.finalColor.b);
+
+            private void IncrementCounter(HDGpuLightsBuilder.GPULightTypeCountSlots counterSlot)
+            {
+                unsafe
+                {
+                    int* ptr = (int*)gpuLightCounters.GetUnsafePtr<int>() + (int)counterSlot;
+                    Interlocked.Increment(ref UnsafeUtility.AsRef<int>(ptr));
+                }
+            }
+
+            public static void ConvertLightToGPUFormat(
```

Comment:
For review sake, is it safe to assume the code this is essentially a move of the code previously in LightLoop.cs? Or is it better to look at the full code again?

Reply:
Its a bit more than that. I will chat with you tomorrow over to look structurally at what was done.